cadc-data-ops-fits/src/main/java/org/opencadc/fits/FitsOperations.java

```diff
@@ -156,20 +159,21 @@ public void headersToStream(final OutputStream outputStream) throws IOException
     /**
      * Implement prototype SODA pixel cutout action.
      *
-     * @param slices subsets of data to extract
+     * @param cutout cutout spec
      * @param outputStream  The Stream to write out to.
-     * @throws ReadException    Any errors to report back to the caller.
+     * @throws ReadException        Any errors reading reported by the storage system.
+     * @throws NoOverlapException   Invalid cutout as there is no match (overlap).
      */
-    public void cutoutToStream(final List<ExtensionSlice> slices, final OutputStream outputStream) throws ReadException {
+    public void cutoutToStream(final Cutout cutout, final OutputStream outputStream) throws ReadException {
+        log.debug("cutoutToStream() start.");
```

Comment:
There should be a check for not-null args and maybe valid Cutout here at the entry point. (just because this is a general purpose library someone could use outside alma and minoc)

See https://github.com/opencadc/dal/issues/68 for a way to make validity checking a one-liner.

Reply:
The requirements in #68 could be met by what the new `SodaCutout` inner class will look like in the next PR for Minoc's `GetAction`:
https://github.com/at88mph/storage-inventory/blob/cadc_8786/minoc/src/main/java/org/opencadc/minoc/GetAction.java

Adding a `validate()` method can simple ensure Its `getConflicts()` method will return an empty `List`.  In fact, much of that `SodaCutout` class could go into this `Cutout` class.